### PR TITLE
allow allies attack

### PR DESCRIPTION
**Pull request by shprotru**
question from chat:
```
Thanks, hello. I have a question about computer allies. In fheroes2 I can't attack them but it's possible in the original game. The map Ghost Planet becomes unplayable if I can't attack allies. Maybe this is an oversight? Or is there a way to force attack anyway?
```
tested with suggested map `Ghost Planet`

### CODE REVIEW

File: src/fheroes2/heroes/heroes_action.cpp [hunk 1]
```diff
@@ -724,10 +725,30 @@ void ActionToHeroes( Heroes & hero, int32_t dst_index )
...
+              && fheroes2::showMessage( fheroes2::Text{ _ ( "Are you sure you want to attack your teammate?" ), fheroes2::FontType::normalWhite() },
+                 fheroes2::Text{ "", fheroes2::FontType::normalWhite() }, Dialog::YES | Dialog::NO )
+                     == Dialog::YES ) {
+        DEBUG_LOG( DBG_GAME, DBG_INFO, hero.GetName() << " attack ally hero " << other_hero->GetName() )
+
+        auto p1 = Players::Get( hero.GetColor() );
+        auto p2 = Players::Get( other_hero->GetColor() );
+
+        at_exit = std::shared_ptr<void>( nullptr,
+                                         std::bind(
```
[the remark this answers is not in the document]
Comment (shprotru) [reply src/fheroes2/heroes/heroes_action.cpp:743]: how to avoid this here

**Comment by ihhub** [comment]
Hi @shprotru , if we are going to implement this logic we **must** make sure that AI will take advantage of this too.

**Comment by shprotru** [comment]
> Hi @shprotru , if we are going to implement this logic we **must** make sure that AI will take advantage of this too.

It would be too much, if AI could betray you

**Comment by zenseii** [comment]
> > Hi @shprotru , if we are going to implement this logic we **must** make sure that AI will take advantage of this too.
> 
> It would be too much, if AI could betray you

Personally I'm very intrigued by this idea :P It would create a lot of replayability if done well. A sort of dynamic alliance system.

**Comment by shprotru** [comment]
Hi, @zenseii it could be controllable in the expected way for a player, like map creator could set betrayal triggers on different events: attack on neutrals, capture object or incursions into some area.

File: src/fheroes2/heroes/heroes_action.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/heroes/heroes_action.cpp:743]: Use a lambda function over std::bind as suggested by clang-tidy. We do not use std::bind in our code.

**Comment by ihhub** [comment]
To proceed with this pull request #5793 must be fixed first as without alliance information it's impossible to know who your enemies are.

**Comment by ihhub** [comment]
This pull request should have a huge revamp in logic and UI. In order to properly support allies to attack each other we need the following:
- add a "weak alliance" feature / option within the Editor for maps to inform users about it
- add an icon and explanation about the alliance within map description dialog
- add description within the Adventure Map scenario information about this logic
- make sure that after attacking any ally all of them are going to turn against you (stop sharing fog discovery) and a new alliance would be formed against the player and against the original enemy
- add logic that the AI can also attack an ally

Based on this, I am closing this pull request as it needs more preparations in terms of overall design for the idea and also much more logic.

**Pull request closed without merging**